aten/src/ATen/native/Convolution.cpp

```diff
@@ -644,6 +644,23 @@ at::Tensor _convolution(
                                       params.padding, params.stride, params.dilation, params.groups);
     }
 #endif
+  } else if (input.device().type() == c10::DeviceType::HAMMERBLADE) {
+    TORCH_CHECK(input.type() == weight.type(),
+             "Input type (", input.type().toString(), ") and weight type (", weight.type().toString(),
+             ") should be the same");
+    TORCH_CHECK(!bias.defined() || (input.type() == bias.type()),
+             "Input type (", input.type().toString(), ") and bias type (", bias.type().toString(),
+             ") should be the same");
+    
+    if (params.transposed) {
+      output = at::hb_convolution_transpose(
```

Comment:
It seems to me that we can directly call `at::native::hb_convolution_transpose` instead of going through dynamic dispatching?